osgar/lib/test_quaternion.py

```diff
@@ -81,4 +81,98 @@ def test_angle_between(self):
         b = [1.7878712250642573e-05, 1.7878701928288437e-05, 3.354137256707306e-13, 0.9999999997613518]
         quaternion.angle_between(a, b)
 
+    def test_slerp__border_cases(self):
+        A = [random.random() for i in range(4)]
+        B = [random.random() for i in range(4)]
+        norm_A = math.sqrt(sum(value**2 for value in A))
+        norm_B = math.sqrt(sum(value**2 for value in B))
+        for i in range(4):
+            A[i] /= norm_A
+            B[i] /= norm_B
+        C = quaternion.slerp(A, B, 0.0)
+        D = quaternion.slerp(A, B, 1.0)
+        for i in range(4):
+            self.assertAlmostEqual(A[i], C[i], 3)
+            self.assertAlmostEqual(B[i], D[i], 3)
+
+    def test_slerp__various_angles(self):
+        sqrt_3 = math.sqrt(3)
+        sqrt_1_7 = math.sqrt(1/7)
+        sqrt_2_7 = math.sqrt(2/7)
+        sqrt_4_7 = math.sqrt(4/7)
+        for a in range(40):
+            angle_A = 10*math.pi*(a - 20)/40
+            sin_A_half = math.sin(angle_A/2)
+            cos_A_half = math.cos(angle_A/2)
+            for b in range(10):
+                angle_B = angle_A + math.pi*(b - 5)/10
+                sin_B_half = math.sin(angle_B/2)
+                cos_B_half = math.cos(angle_B/2)
+                # around x
+                A = [sin_A_half, 0.0, 0.0, cos_A_half]
+                B = [sin_B_half, 0.0, 0.0, cos_B_half]
+                for c in range(10):
```

Comment:
is it really necessary to do these 4000 cycles? I suppose it is trying to cover "all cases"?